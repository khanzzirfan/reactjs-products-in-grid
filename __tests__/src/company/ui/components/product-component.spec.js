import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ProductComponent from 'src/products/ui/components/product-component';

describe("Product Component test", () => {

    let stubDataProps = {
        product: {
            "id": "37622-2oagdixqdw3mcf20wudlwyiudi",
            "size": 35,
            "price": 149,
            "face": "(⊙ω⊙)",
            "date": "Sat Feb 10 2018 17:23:46 GMT+1300 (New Zealand Daylight Time)"
        }
    };

    let wrapper = null;
    beforeEach(() => {
        wrapper = shallow(<ProductComponent {...stubDataProps} />);
    })

    it("component should be defined", () => {
        expect(wrapper).toBeDefined();
    });

    it("component should match to snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it(" should render component with correct props", () => {
        expect(wrapper).toMatchSnapshot();
        
        expect(wrapper.instance().props.product.id).toEqual("37622-2oagdixqdw3mcf20wudlwyiudi");
        expect(wrapper.instance().props.product.size).toEqual(35);
        expect(wrapper.instance().props.product.price).toEqual(149);
        expect(wrapper.instance().props.product.face).toEqual("(⊙ω⊙)");
        expect(wrapper.instance().props.product.date).toEqual("Sat Feb 10 2018 17:23:46 GMT+1300 (New Zealand Daylight Time)");
    });

    it("should render the product face once and correct size from props", () => {
        let faceDiv = wrapper.find({ id: "product-face" });
        expect(faceDiv.length).toEqual(1);
        expect(faceDiv.text()).toEqual("(⊙ω⊙)");
        expect(faceDiv.prop('style')).toEqual({fontSize:'35px'});
    });

});