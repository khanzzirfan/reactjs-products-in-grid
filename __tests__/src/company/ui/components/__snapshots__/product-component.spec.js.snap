// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component test  should render component with correct props 1`] = `
<div
  className="col-lg-3 col-md-3 col-sm-6 portfolio-item"
>
  <div
    className="card h-100"
  >
    <div
      className="face"
      id="product-face"
      style={
        Object {
          "fontSize": "35px",
        }
      }
    >
      (⊙ω⊙)
    </div>
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        id: 
        37622-2oagdixqdw3mcf20wudlwyiudi
         
      </p>
      <p
        className="card-text"
      >
         size: 
        35
      </p>
      <p
        className="card-text"
      >
        price: $
        149
         
      </p>
      <p
        className="card-text"
      >
        date: 
        Sat Feb 10 2018 17:23:46 GMT+1300 (New Zealand Daylight Time)
         
      </p>
    </div>
  </div>
</div>
`;

exports[`Product Component test component should match to snapshot 1`] = `
<div
  className="col-lg-3 col-md-3 col-sm-6 portfolio-item"
>
  <div
    className="card h-100"
  >
    <div
      className="face"
      id="product-face"
      style={
        Object {
          "fontSize": "35px",
        }
      }
    >
      (⊙ω⊙)
    </div>
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        id: 
        37622-2oagdixqdw3mcf20wudlwyiudi
         
      </p>
      <p
        className="card-text"
      >
         size: 
        35
      </p>
      <p
        className="card-text"
      >
        price: $
        149
         
      </p>
      <p
        className="card-text"
      >
        date: 
        Sat Feb 10 2018 17:23:46 GMT+1300 (New Zealand Daylight Time)
         
      </p>
    </div>
  </div>
</div>
`;
